consensus/core/src/stake_aggregator.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use consensus_config::{AuthorityIndex, Committee, Stake};
+use std::collections::HashSet;
+use std::marker::PhantomData;
+
+pub trait CommitteeThreshold {
+    fn is_threshold(committee: &Committee, amount: Stake) -> bool;
+}
+
+pub(crate) struct QuorumThreshold;
+
+pub(crate) struct ValidityThreshold;
+
+impl CommitteeThreshold for QuorumThreshold {
+    fn is_threshold(committee: &Committee, amount: Stake) -> bool {
+        committee.reached_quorum(amount)
+    }
+}
+
+impl CommitteeThreshold for ValidityThreshold {
+    fn is_threshold(committee: &Committee, amount: Stake) -> bool {
+        committee.reached_validity(amount)
+    }
+}
+
+#[allow(unused)]
+
+pub struct StakeAggregator<T> {
+    votes: HashSet<AuthorityIndex>,
```

Comment:
Any reason you decided not to use the `AuthoritySet` implementation from mysticeti?

Reply:
resolved offline